#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.5617427335642976 -0.6218859737343058 -0.5456216060240329 1.9333935769738921
  position 0.6315931402245627 2.940414228787209 0.7055519170592357
  follow "robot:distance_sensor_front"
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF GROUND Group {
  children [
    Solid {
      translation 2.1 0 0
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/RT.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid7"
    }
    Solid {
      translation 2.1 0 0.3
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2R.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid72"
    }
    Solid {
      translation 2.1 0 0.6
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2R.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid73"
    }
    Solid {
      translation 2.1 0 0.9
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2R.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid74"
    }
    Solid {
      translation 2.1 0 1.2
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2R.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid75"
    }
    Solid {
      translation 2.1 0 1.5
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2R.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid76"
    }
    Solid {
      translation 2.1 0 1.8
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/RB.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid77"
    }
    Solid {
      translation 1.8 0 0
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1T.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid6"
    }
    Solid {
      translation 1.8 0 0.3
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid62"
    }
    Solid {
      translation 1.8 0 0.6
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid63"
    }
    Solid {
      translation 1.8 0 0.9
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid64"
    }
    Solid {
      translation 1.8 0 1.2
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid65"
    }
    Solid {
      translation 1.8 0 1.5
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid66"
    }
    Solid {
      translation 1.8 0 1.8
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1B.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid67"
    }
    Solid {
      translation 1.5 0 0
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1T.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid5"
    }
    Solid {
      translation 1.5 0 0.3
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid52"
    }
    Solid {
      translation 1.5 0 0.6
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid53"
    }
    Solid {
      translation 1.5 0 0.9
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid54"
    }
    Solid {
      translation 1.5 0 1.2
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid55"
    }
    Solid {
      translation 1.5 0 1.5
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid56"
    }
    Solid {
      translation 1.5 0 1.8
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1B.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid57"
    }
    Solid {
      translation 1.2 0 0
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1T.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid4"
    }
    Solid {
      translation 1.2 0 0.3
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid42"
    }
    Solid {
      translation 1.2 0 0.6
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid43"
    }
    Solid {
      translation 1.2 0 0.9
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/20.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid44"
    }
    Solid {
      translation 1.2 0 1.2
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid45"
    }
    Solid {
      translation 1.2 0 1.5
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid46"
    }
    Solid {
      translation 1.2 0 1.8
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1B.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid47"
    }
    Solid {
      translation 0.9 0 0
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1T.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid3"
    }
    Solid {
      translation 0.9 0 0.3
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid32"
    }
    Solid {
      translation 0.9 0 0.6
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid33"
    }
    Solid {
      translation 0.9 0 0.9
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid34"
    }
    Solid {
      translation 0.9 0 1.2
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid35"
    }
    Solid {
      translation 0.9 0 1.5
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid36"
    }
    Solid {
      translation 0.9 0 1.8
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1B.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid37"
    }
    Solid {
      translation 0.3 0 0
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/LT.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid1"
    }
    Solid {
      translation 0.3 0 0.3
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2L.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid12"
    }
    Solid {
      translation 0.3 0 0.6
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2L.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid13"
    }
    Solid {
      translation 0.3 0 0.9
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2L.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid14"
    }
    Solid {
      translation 0.3 0 1.2
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2L.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid15"
    }
    Solid {
      translation 0.3 0 1.5
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/2L.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid16"
    }
    Solid {
      translation 0.3 0 1.8
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/LB.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid17"
    }
    Solid {
      translation 0.6 0 0
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1T.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid2"
    }
    Solid {
      translation 0.6 0 0.3
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid22"
    }
    Solid {
      translation 0.6 0 0.6
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid23"
    }
    Solid {
      translation 0.6 0 0.9
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/10.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid24"
    }
    Solid {
      translation 0.6 0 1.2
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid25"
    }
    Solid {
      translation 0.6 0 1.5
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/0.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid26"
    }
    Solid {
      translation 0.6 0 1.8
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../tiles/1B.png"
              ]
            }
          }
          geometry Plane {
            size 0.3 0.3
          }
        }
      ]
      name "solid27"
    }
    Solid {
      translation 1.25 -0.049 1
      boundingObject Box {
        size 2.5 0.1 2.5
      }
    }
  ]
}
DEF LIGHTSOURCE Group {
  children [
    PointLight {
      attenuation 0 0 1
      color 0 0 1
      intensity 0.1
      location 1.1 0.1 0.9
    }
  ]
}
DEF WALLS Group {
  children [
    Wall {
      translation 1.2 0 0.75
      name "wall1"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 1.2 0 1.05
      name "wall2"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 1.05 0 0.9
      name "wall3"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 0.75 0 0
      name "wall4"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 1.65 0 1.8
      name "wall7"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 1.65 0 0
      name "wall6"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 0.75 0 1.8
      name "wall5"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 1.65 0 0.6
      name "wall8"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 1.65 0 1.2
      name "wall9"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 2.1 0 0.75
      name "wall10"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 1.8 0 1.05
      name "wall11"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 1.5 0 0.75
      name "wall12"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 1.05 0 1.2
      name "wall13"
      size 0.05 0.5 0.3
    }
    Wall {
      translation 1.5 0 1.35
      name "wall14"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 0.9 0 0.45
      name "wall15"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 0.9 0 0.15
      name "wall16"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 1.2 0 0.15
      name "wall17"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 0.3 0 0.45
      name "wall18"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 0.3 0 1.35
      name "wall19"
      size 0.3 0.5 0.05
    }
    Wall {
      translation 0.6 0 1.35
      name "wall20"
      size 0.3 0.5 0.05
    }
  ]
}
Robot {
  translation 0.918166 0.0424467 0.913267
  rotation 0.008505708494184132 0.9999638229704918 -7.532098666548037e-05 1.58995
  children [
    InertialUnit {
      name "intertialUnit"
    }
    GPS {
    }
    Compass {
    }
    DistanceSensor {
      translation 0 -0.03 0.055
      rotation 0 0 1 -1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "line_sensor"
      physics Physics {
      }
      type "infra-red"
    }
    LightSensor {
      translation 0 0.02 0.05
      rotation 0 1 0 -1.59
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "light_sensor"
      colorFilter 0 0 1
      occlusion TRUE
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 21726.487823292544
        anchor -0.02 -0.02 0.03
      }
      device [
        RotationalMotor {
          name "FRWHEEL"
        }
      ]
      endPoint Solid {
        translation -0.049999999999998476 -0.02 0.03
        rotation -0.6085660842384079 0.6085683196305575 -0.5092071498484392 4.083562022523314
        children [
          Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.02
            }
          }
        ]
        boundingObject Cylinder {
          height 0.02
          radius 0.02
        }
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 21740.479080688885
        anchor -0.02 -0.02 -0.03
      }
      device [
        RotationalMotor {
          name "BRWHEEL"
        }
      ]
      endPoint Solid {
        translation -0.05000000000000064 -0.02 -0.03
        rotation -0.6071557992074039 0.6071580294192778 -0.5125631305512696 4.08888472418462
        children [
          Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.02
            }
          }
        ]
        name "solid(2)"
        boundingObject Cylinder {
          height 0.02
          radius 0.02
        }
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 31902.269639343467
        anchor 0.015 -0.02 -0.03
      }
      device [
        RotationalMotor {
          name "BLWHEEL"
        }
      ]
      endPoint Solid {
        translation 0.049999999999999226 -0.02 -0.03
        rotation -0.7054291415684192 0.7054317327591011 0.06881712463141376 3.0041755614886445
        children [
          Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.02
            }
          }
        ]
        name "solid(3)"
        boundingObject Cylinder {
          height 0.02
          radius 0.02
        }
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 31893.259215216072
        anchor 0.015 -0.02 0.03
      }
      device [
        RotationalMotor {
          name "FLWHEEL"
        }
      ]
      endPoint Solid {
        translation 0.04999999999999951 -0.02 0.03
        rotation -0.7056043097743127 0.7056069016084163 0.0651264802556263 3.011523856441064
        children [
          Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject Cylinder {
          height 0.02
          radius 0.02
        }
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 6.99334e-05 0.00982561 0.0493899
      rotation 0.01112216575729272 -0.9999129210601427 -0.007102656252445453 1.5776432937483473
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.02
          }
        }
      ]
      name "distance_sensor_front"
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.05 0.0109391 -0.000260794
      rotation 0 -1 0 -5.307179586466759e-06
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.02
          }
        }
      ]
      name "distance_sensor_left"
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.000742173 0.0116556 -0.0505911
      rotation 0 1 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.02
          }
        }
      ]
      name "distance_sensor_back"
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.0492576 0.0116556 -0.0005911
      rotation 0 1 0 -3.1415853071795863
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.02
          }
        }
      ]
      name "distance_sensor_right"
      physics Physics {
      }
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.1
      }
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "inz_controller_py"
}
